apps/site/assets/ts/tnm/components/RouteCard.tsx

```diff
@@ -46,7 +69,12 @@ const RouteCard = ({
 
   return (
     <div className="m-tnm-sidebar__route" data-mode={mode}>
-      <RouteCardHeader route={route.route} alerts={route.route.alerts} />
+      <RouteCardHeader
+        route={route.route}
+        alerts={isDefined(
+          useSWR<Alert[]>(`${route.route.id}`, fetchAlerts).data
+        )}
```

Comment:
suggestion: this is nicely condensed, but sacrifices readability a bit. we can use some ES6 syntax ([object destructuring](https://javascript.info/destructuring-assignment)) to help us a bit here:

```typescript
// destructures the "data" property out of the response object and renames it to "alerts"
// alerts might be undefined here
const { data: alerts } = useSWR<Alert[]>(`${route.route.id}`, fetchAlerts);
```

Also you can assign a fallback value if the initial value is false with `||`. So you could pass in `alerts={alerts || []}` and it should work in the same manner as your `isDefined` function.